### PR TITLE
fix: Properties had a null description

### PR DESCRIPTION
**Pull request by Blu-J**
### About

During the current js version of the proxy, getting an error in the properties because it is failing to check the shape

Looks like the issue was that the value was coming back but as null, and I wasn't dealing with the null case, just the undefined case.

### Proof Of Fix 

Picture of the properties on the proxy working now

image

### CODE REVIEW

File: frontend/projects/ui/src/app/util/properties.util.ts [hunk 1]
```diff
@@ -149,12 +149,12 @@ function parsePropertiesV1Permissive(
       } else {
         const error = result.error
         const message = Parser.validatorErrorAsString(error)
-        let dataPath = error.keys.map(x => JSON.parse(x)).join('/')
+        const dataPath = '/' + error.keys.map(removeQuotes).join('/')
```
Comment (Blu-J) [inline frontend/projects/ui/src/app/util/properties.util.ts:152]:
```suggestion
        const dataPath = error.keys.map(removeQuotes).join('/')
```

File: frontend/projects/ui/src/app/util/properties.util.ts [hunk 2]
```diff
@@ -41,11 +41,11 @@ const [matchPackagePropertiesV2, setPPV2] = deferred<PackagePropertiesV2>()
...
     type: literal('string'),
-    description: string,
+    description: string.optional(),
     value: string,
-    copyable: boolean,
-    qr: boolean,
-    masked: boolean,
+    copyable: boolean.optional(),
+    qr: boolean.optional(),
+    masked: boolean.optional(),
```
Comment (Blu-J) [inline frontend/projects/ui/src/app/util/properties.util.ts:48]: I'm going to probably make a fix for the ts-matches about this...

File: frontend/projects/ui/src/app/util/properties.util.ts [hunk 3]
```diff
@@ -149,12 +149,12 @@ function parsePropertiesV1Permissive(
...
         const error = result.error
         const message = Parser.validatorErrorAsString(error)
-        let dataPath = error.keys.map(x => JSON.parse(x)).join('/')
+        const dataPath = error.keys.map(removeQuotes).join('/')
         errorCallback(new Error(`/data/${idx}: ${message}`))
         if (dataPath) {
           applyOperation(cur, {
             op: 'replace',
-            path: dataPath,
+            path: `/${dataPath}`,
```
Comment (Blu-J) [inline frontend/projects/ui/src/app/util/properties.util.ts:157]: I probably shouldn't make this change. Paths should start with /
Just noticed it was wrong when looking at this

File: frontend/projects/ui/src/app/util/properties.util.ts [hunk 4]
```diff
@@ -196,6 +196,14 @@ function parsePropertiesV2Permissive(
   )
 }
 
+const removeRegex = /('|")/
+function removeQuotes(x: string) {
```
Comment (Blu-J) [inline frontend/projects/ui/src/app/util/properties.util.ts:200]: image

File: frontend/projects/ui/src/app/util/properties.util.ts (hunk 2 above)
Comment (Blu-J) [reply frontend/projects/ui/src/app/util/properties.util.ts:48]: https://github.com/Blu-J/ts-matches/issues/505

File: frontend/projects/ui/src/app/util/properties.util.ts (hunk 3 above)
Comment (elvece) [reply frontend/projects/ui/src/app/util/properties.util.ts:157]: No this is fine as it it complied with the specification

**Review by elvece — APPROVED** [review]

**Pull request merged**